server/dash/dashboards/neighborhood.py

```diff
@@ -39,27 +42,80 @@ def populate_options():
         )
     return values
 
+print(populate_options())
+
+layout = html.Div([
+    html.H1(title),
+    dcc.Dropdown(
+        id='council_list',
+        clearable=False,
+        value="Arleta",
+        placeholder="Select a neighborhood",
+        options=populate_options()
+    ),
+    dcc.Dropdown(
+        id='data_type',
+        clearable=False,
+        placeholder="Select a timeframe",
+        options=[
+            {'label': 'Compare to: Daily Council Average', 'value': 'nc_avg'},
+            {'label': 'Compare to: Weekly Council Average', 'value': 'nc_avg_wk'},
+            {'label': 'Compare to: Monthly Council Average', 'value': 'nc_avg_Mo'},
+        ],
+
+        value='nc_avg'
+    ),
+    dcc.Graph(
+        id='graph1',
+        figure=fig,
+        config=CONFIG_OPTIONS
+    ),
+
+    dcc.Graph(
+        id='graph2',
+        figure=fig,
+        config=CONFIG_OPTIONS
+    )
+])
+
 
 # Define callback to update graph
 @app.callback(
     Output('graph1', 'figure'),
-    [Input("council_list", "value")]
+    [Input("council_list", "value"),
+    Input("data_type", "value")]
 )
-def update_figure(selected_council):
-
+def update_figure(selected_council, select_timeframe):
     neighborhood_sum_df = df[df.council_name == selected_council].groupby(['created_date']).agg('sum').reset_index()  # noqa
+
     total_sum_df = df.groupby(['created_date']).agg('sum').reset_index()
+
     total_sum_df["nc_avg"] = total_sum_df["counts"] / 99
-    merged_df = neighborhood_sum_df.merge(total_sum_df["nc_avg"].to_frame(), left_index=True, right_index=True)  # noqa
 
+    total_sum_df["week_n"] = total_sum_df['created_date'].apply(lambda x: str(x.isocalendar()[1]))
+    total_sum_df["month_n"] = total_sum_df['created_date'].dt.month
+
+    total_sum_df['nc_avg_Mo'] = total_sum_df.groupby(['month_n'])['counts'].transform('mean')
+    total_sum_df['nc_avg_Mo'] = total_sum_df['nc_avg_Mo'] / 99
+    total_sum_df['nc_avg_wk'] = total_sum_df.groupby(['week_n'])['counts'].transform('mean')
+    total_sum_df['nc_avg_wk'] = total_sum_df['nc_avg_wk'] / 99
+    merged_df = pd.merge(neighborhood_sum_df, total_sum_df, on=["created_date", "created_date"])
+    
+
+    
     fig = px.line(
         merged_df,
         x="created_date",
-        y=['counts', 'nc_avg'],
+        y=['counts_x', select_timeframe],
         color_discrete_sequence=DISCRETE_COLORS,
-        labels=LABELS,
         title="Comparison trend for " + selected_council
     )
+    newnames = {"counts_x": "Count of 311 Requests", "nc_avg_Mo": "Monthly Council Average", "nc_avg_wk": "Weekly Council Average", "nc_avg": "Daily Council Average"}
```

Comment:
We can pull out the nc_avg* keys into a separate dict and use that both for renaming here and the dropdown above. That way, if we want to change the names, we only need to do it in one place.

Reply:
I have changed the way the code works. Now it might be closer to what you suggested.